lsm-tree.cabal

```diff
@@ -52,45 +52,103 @@ description:
+  *   The space complexity of tables is described in terms of the in-memory size of an LSM-tree table.
+      Both the in-memory and on-disk size of an LSM-tree table scales linearly.
+      However, the in-memory size of an LSM-tree table is smaller than its on-disk size by a significant constant.
+      This is discussed in detail below, under [In-memory size of tables](#performance_size).
+
+  The complexities are described in terms of the following variables and constants:
+
+  *   The variable \(n\) refers to the number of /physical/ table entries.
+      A /physical/ table entry is any key–operation pair, e.g., @Insert k v@ or @Delete k@, whereas a /logical/ table entry is determined by all physical entries with the same key.
+      If the variable \(n\) is used to describe the complexity of an operation that involves multiple tables, it refers to the sum of all table entries.
+  *   The constant \(T\) refers to the size ratio of the table, which is a configuration parameter.
+
+  === Disk I\/O cost of operations #performance_time#
+
+  The following table summarises the cost of the operations on LSM-trees
+  measured in the number of disk I\/O operations.
+  If the cost depends on the merge policy, the table contains one entry for each merge policy.
+  Otherwise, the merge policy is listed as N\/A.
+
+  +----------+------------------------+----------------------------+------------------------------+
+  | Resource | Operation              | Merge policy               | Cost in disk I\/O operations |
+  +==========+========================+============================+==============================+
+  | Session  | Create\/Open           | N\/A                       | \(O(1)\)                     |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Close                  | N\/A                       | \(O(t \: \log_T \: n)\)*     |
+  +----------+------------------------+----------------------------+------------------------------+
+  | Table    | Create                 | N\/A                       | \(O(1)\)                     |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Close                  | N\/A                       | \(O(\log_T \: n)\)           |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Lookup                 | @MergePolicyLazyLevelling@ | \(O(\log_T \: n)\)           |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Range Lookup           | N\/A                       | \(O(\log_T \: n + b)\)**     |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Insert\/Delete\/Update | @MergePolicyLazyLevelling@ | \(O(\log_T \: n)\)           |
```

Comment:
I'm not sure if simplification has done the complexities justice. We're omitting bloom filter false positive rates, write buffer size, number of entries per page, page size, distribution of inserted keys, etc., and that makes the bounds very loose. 

 I'm currently leaning towards suggesting to make these complexity bounds detailed rather than simple, following the Monkey paper more closely. What are your thoughts about this?